core/src/main/java/org/apache/accumulo/core/clientImpl/ConditionalWriterImpl.java

```diff
@@ -801,6 +831,8 @@ public Result write(ConditionalMutation mutation) {
   @Override
   public void close() {
     threadPool.shutdownNow();
+    threadPoolCleanable.clean(); // deregister the cleaner, will not call shutdownNow() because
+                                 // closed is now true
     cleanupThreadPool.execute(Threads.createNamedRunnable("ConditionalWriterCleanupTask",
```

Comment:
I this code was written w/ the assumption that there was a static thread pool where we could submit a task that reaches out to tservers and does some optional clean up.  Now that thread pool is not static, using this thread pool seems to complicate things.  We could create a daemon thread to do this `new Thread(new CleanupTask(getActiveSessions())).setDeamon(true).run()` instead of submitting a task to the thread pool.  That way the thread pool could closed. When the runnable finishes I think the thread will just go away.

Also do you know if the cleaner actually shuts the thread pool down?  I wonder if the threads in a thread pool can reference the thread pool object so it always looks referenced from the GC perspective.

Reply:
Addressed in 918b20f. CleanupThreadPool is now a reference to the shared scheduled executor pool.